### ISSUE TITLE
collapse_isoforms_precise crashing

### ISSUE THREAD

**Issue by diekhans** [BrooksLabUCSC/flair#68]
I am running flair collapse with the --isoformtss --no_redundant=longest options, which results in collapse_isoforms_precise.py crashing with

`+ python2 /hive/users/markd/gencode/projs/icedb/flair-markd/bin/collapse_isoforms_precise.py -q correct_all_corrected.psl -m 2 -w 100 -n longest -o collapse.firstpass.unfiltered.psl -i
Read data extracted
Single-exon genes grouped, collapsing
Traceback (most recent call last):
  File "/hive/users/markd/gencode/projs/icedb/flair-markd/bin/collapse_isoforms_precise.py", line 684, in <module>
    res = p.map(run_find_best_sites, list(isoforms.keys()))
  File "/cluster/home/markd/opt/centos7.0/x86_64/lib/python2.7/multiprocessing/pool.py", line 253, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/cluster/home/markd/opt/centos7.0/x86_64/lib/python2.7/multiprocessing/pool.py", line 572, in get
    raise self._value
TypeError: 'int' object has no attribute '__getitem__'
`

Attached are the files to reproduce this
[collapse-bug.tar.gz](https://github.com/BrooksLabUCSC/flair/files/3779731/collapse-bug.tar.gz)

**Comment by belgravia** [comment]
Thanks for the files and command.
There was a bug having to do with running flair collapse with `-n longest` that I have addressed with the most recent commit. 

I used the command `python ~/ba/flair/bin/collapse_isoforms_precise.py -q correct_all_corrected.psl -m 2 -w 100 -n longest -i -o collapse.firstpass.unfiltered.psl` with python 2.7 and I have attached the resulting file.
[collapse.firstpass.unfiltered.psl.gz](https://github.com/BrooksLabUCSC/flair/files/3780169/collapse.firstpass.unfiltered.psl.gz)

**Comment by diekhans** [comment]
thanks Alison, however, the PSL part of the file contains some large negative numbers:


> 1017	0	0	0	0	0	1	585	+	782a5147-b80d-488c-b585-ff15119b2bcd;0	1017	0	1017	chr19_KI270867v1_alt	233762	130876	1	2	993,-132453,	0,993,	130876,132454,	1.0

**Comment by belgravia** [comment]
Thanks for your patience. I fixed the large negative numbers issue
[correct_all_corrected.collapsed.psl.gz](https://github.com/BrooksLabUCSC/flair/files/3780691/correct_all_corrected.collapsed.psl.gz)
. While I was fixing the issue earlier today, I accidentally changed which item in a list was being indexed so I changed that back now.

I think I should also mention that this issue is a separate issue from the `-1` blocksizes at the end, I am still not sure what the source of that issue is.
-Alison
